crates/dht/src/dht.rs

```diff
@@ -346,9 +346,16 @@ impl DhtApi for Dht {
                 SnapshotDiff::DiscSectorSliceMismatches(
                     mismatched_slice_indices,
                 ) => {
+                    // Need to fetch in order by sector index
```

Comment:
This wasn't exactly a bug but it was a little odd. We were letting the hashmap decide whatever to return data in. The tests I'd written were flaky like this once I split the op data over multiple slices.

Reply:
So when retrieving all the data in one go, no problem. In terms of syncing everything from a large DHT and needing all of it. Sure we'll get it eventually. It makes a lot more sense to me for this to be predictable.